app/main.py

```diff
@@ -1,2 +1,148 @@
+            capacity_multiplier: int | float = _DEFAULT_CAPACITY_MULTIPLIER
+    ) -> None:
+        self._capacity = initial_capacity
+        self._load_factor = load_factor
+        self._threshold = int(initial_capacity * load_factor)
+        self._capacity_multiplier = capacity_multiplier
+        self._size = 0
+        self._hash_table: list[Dictionary.Node | None] = (
+            [None] * self._capacity
+        )
+
+    def _calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self._capacity
+
+        while (
+                self._hash_table[index] is not None
+                and self._hash_table[index].key != key
+        ):
+            index = (index + 1) % self._capacity
+
+        return index
+
+    def __getitem__(self, key: Hashable) -> Any:
+        item: Dictionary.Node | None = (
+            self._hash_table[self._calculate_index(key)]
+        )
+
+        if not item:
+            raise KeyError(f"'{key}'")
+
+        return item.value
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self._calculate_index(key)
+
+        if not self._hash_table[index]:
+            if self._size + 1 >= self._threshold:
+                self._resize_hash_table()
+                return self.__setitem__(key, value)
+            self._hash_table[index] = Dictionary.Node(key, value)
+            self._size += 1
+        else:
+            self._hash_table[index].value = value
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = self._calculate_index(key)
+
+        if not self._hash_table[index]:
+            raise KeyError(f"'{key}'")
+
+        self._hash_table[index] = None
+        self._size -= 1
+
+    def __iter__(self) -> Dictionary:
+        self._current_iteration = 0
+        return self
+
+    def __next__(self) -> (Hashable, Any):
+        while self._current_iteration < self._capacity:
+            item = self._hash_table[self._current_iteration]
+            self._current_iteration += 1
+            if item:
+                return item.key, item.value
+
+        raise StopIteration
+
+    def __len__(self) -> int:
+        return self._size
+
+    def __str__(self) -> str:
+        return str(self._hash_table)
+
+    def _resize_hash_table(self) -> None:
+        previous_hash_table = self._hash_table
+
+        self.__init__(
+            self._capacity * self._capacity_multiplier,
+            self._load_factor,
+            self._capacity_multiplier
+        )
+
+        for node in previous_hash_table:
+            if node:
+                self[node.key] = node.value
+
+    def clear(self) -> None:
+        self._size = 0
+        self._hash_table = [None] * self._capacity
+
+    def get(self, key: Hashable, default_value: Any = None) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            return default_value
+
+    def pop(self, key: Hashable, default_value: Any = None) -> Any:
```

Comment:
pop method works a little different it will trigger a keyerror if key is not in dict and default values is not set